File: packages/server-php/src/SessionConfig.php
<?php

declare(strict_types=1);

namespace OpenPasskey\Server;

class SessionConfig
{
    public function __construct(
        public readonly string $secret,
        public readonly int $durationSeconds = 86400,
        public readonly int $clockSkewGraceSeconds = 10,
        public readonly string $cookieName = 'op_session',
        public readonly string $cookiePath = '/',
        public readonly bool $secure = true,
        public readonly string $sameSite = 'Lax',
        public readonly ?string $domain = null,
    ) {}
}

File: packages/server-rails/lib/open_passkey/session_config.rb
# frozen_string_literal: true

module OpenPasskey
  class SessionConfig
    attr_accessor :secret, :duration_seconds, :clock_skew_grace_seconds,
                  :cookie_name, :cookie_path, :secure, :same_site, :domain

    def initialize(secret:, duration_seconds: 86400, clock_skew_grace_seconds: 10,
                   cookie_name: "op_session", cookie_path: "/",
                   secure: true, same_site: "Lax", domain: nil)
      @secret = secret
      @duration_seconds = duration_seconds
      @clock_skew_grace_seconds = clock_skew_grace_seconds
      @cookie_name = cookie_name
      @cookie_path = cookie_path
      @secure = secure
      @same_site = same_site
      @domain = domain
    end
  end
end
